db/bigtable.go

```diff
@@ -937,25 +1007,23 @@ func (bigtable *Bigtable) GetValidatorMissedAttestationsCount(validators []uint6
 
 	res := make(map[uint64]*types.ValidatorMissedAttestationsStatistic)
 
-	for e := firstEpoch; e <= lastEpoch; e++ {
-		data, err := bigtable.GetValidatorAttestationHistory(validators, e, e)
+	data, err := bigtable.GetValidatorMissedAttestationHistory(validators, firstEpoch, lastEpoch)
 
-		if err != nil {
-			return nil, err
-		}
+	if err != nil {
+		return nil, err
+	}
 
-		logger.Infof("retrieved attestation history for epoch %v", e)
+	logger.Infof("retrieved attestation history for epochs %v - %v", firstEpoch, lastEpoch)
 
-		for validator, attestations := range data {
-			for _, attestation := range attestations {
-				if attestation.Status == 0 {
-					if res[validator] == nil {
-						res[validator] = &types.ValidatorMissedAttestationsStatistic{
-							Index: validator,
-						}
+	for validator, attestations := range data {
+		for _, missed := range attestations {
+			if missed {
+				if res[validator] == nil {
+					res[validator] = &types.ValidatorMissedAttestationsStatistic{
+						Index: validator,
 					}
-					res[validator].MissedAttestations++
 				}
+				res[validator].MissedAttestations++
 			}
 		}
 	}
```

Comment:
Similar to my comment in `validator.go` for
https://github.com/gobitfly/eth2-beaconchain-explorer/blob/65cb3a34ffa22b5a6c1db0d82daaf2685c80f128/handlers/validator.go#L694-L698

=> Shouldn't it be possible to use `len` here directly?

Reply:
refactored